packages/xod-client/src/editor/containers/Patch/modes/linking.jsx

```diff
@@ -64,7 +69,7 @@ const linkingMode = {
             width={api.props.size.width}
             height={api.props.size.height}
             onClick={bindApi(api, this.onBackgroundClick)}
-            offset={api.props.offset}
+            offset={roundOffsetToPixel(api.props.offset)}
```

Comment:
A lot of error-prone code duplication. I suggest postponing rounding as long as possible, for example, do it once in `BackgroundLayer.jsx` → `NodeSlotPattern`